apps/token/src/routes/staking/node/staking-form.tsx

```diff
@@ -196,43 +197,35 @@ export const StakingForm = ({
     return () => clearInterval(interval);
   }, [formState, client, pubKey, nodeId]);
 
-  if (formState === FormState.Failure) {
-    return <StakeFailure nodeName={nodeName} />;
-  } else if (formState === FormState.Requested) {
-    return (
-      <Dialog
-        title="Confirm transaction in wallet"
-        intent={Intent.Warning}
-        open={true}
-      >
-        <p>{t('stakingConfirm')}</p>
-      </Dialog>
-    );
-  } else if (formState === FormState.Pending) {
-    return <StakePending action={action} amount={amount} nodeName={nodeName} />;
-  } else if (formState === FormState.Success) {
-    return (
-      <StakeSuccess
-        action={action}
-        amount={amount}
-        nodeName={nodeName}
-        removeType={removeType}
-      />
-    );
-  } else if (
-    availableStakeToAdd.isEqualTo(0) &&
-    availableStakeToRemove.isEqualTo(0)
-  ) {
-    if (appState.lien.isGreaterThan(0)) {
-      return <span className="text-red">{t('stakeNodeWrongVegaKey')}</span>;
-    } else {
-      return <span className="text-red">{t('stakeNodeNone')}</span>;
+  const toggleDialog = useCallback(() => {
+    setIsDialogVisible(!isDialogVisible);
+  }, [isDialogVisible]);
+
+  useEffect(() => {
+    if (
+      formState === FormState.Success ||
+      formState === FormState.Failure ||
+      formState === FormState.Pending ||
+      formState === FormState.Requested
+    ) {
+      setIsDialogVisible(true);
     }
-  }
+  }, [formState]);
 
   return (
     <>
       <h2>{t('Manage your stake')}</h2>
+      {formState === FormState.Default &&
+        availableStakeToAdd.isEqualTo(0) &&
+        availableStakeToRemove.isEqualTo(0) && (
+          <div>
+            {appState.lien.isGreaterThan(0) ? (
+              <span className="text-red">{t('stakeNodeWrongVegaKey')}</span>
+            ) : (
+              <span className="text-red">{t('stakeNodeNone')}</span>
+            )}
+          </div>
+        )}
```

Comment:
chore (blocking): this needs to be outside this render, in this case the user should not be able to see the form as they have no stake that is addable.

Reply:
@dexturr The form button doesn't allow the form to be usable unless there's sufficient stake, I thought it was more useful for users to see what they will be able to do, once these messages are resolved.

Reply:
Kewl